fsspec/__init__.py

```diff
@@ -1,15 +1,19 @@
-from ._version import get_versions
+try:
+    from importlib.metadata import entry_points
+except ImportError:  # python < 3.8
+    from importlib_metadata import entry_points
```

Comment:
actually, we should cope with this not being present, since it is not an absolutely required dependency - which I agree with. This, if py<3.8, and importlib_metadata is not present, then setting of the registry via entrypoints should be skipped.

Reply:
We could add test to the error you get if a backend is not found *and* importlib_metadata is not present, to say that installing it may expose some implementations.

Reply:
registry via entrypoints can still be done without this backport package in 3.6 and 3.7 but I felt since this was the method going forward that it would be easier to remove this shim once 3.6 and 3.7 are no longer supported than to implement the "old" entrypoint method, then update to this method later.

How do you feel about just raising a warning if the second import attempt fails? That way it's still possible to operate without it, but is a more conscious choice

Reply:
I'd rather not clutter the output with warnings, but silently ignoring is OK, I think, since *new* backends are surely being developed for py>=3.8. If people raise issues around backends failing to register, we first point to this discussion, then maybe change the behaviour if needed.

Reply:
(I can do this, if you don't have time)

Reply:
Actually, it might be better if you do... I was toying with using the old setup tools iter_entrypoints() method as a final fallback but I can seem to form it in a way that's readable. I can always open another PR if that comes together